tools/github/setup_test_env.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+# Options:
+#
+# MINIMUM_REQUIREMENTS:
+#   Install the minimum versions of all requirements, instead of the latest.
+#
+# OPTIONAL_DEPS:
+#   Install optional requirements.
+#
+# WITHOUT_POOCH:
+#   Remove pooch from environment.
+#
+# PIP_FLAGS:
+#   These options are passed to pip.
+
+# TODO: Remove special handling of free-threaded dependencies below.
+
+set -ex
+
+export PIP_DEFAULT_TIMEOUT=60
+python -m pip install --upgrade pip
+
+if [[ $MINIMUM_REQUIREMENTS == 1 ]]; then
+    for filename in requirements/*.txt; do
+        sed -i 's/>=/==/g' "$filename"
+    done
+fi
```

Comment:
This could also `sed` the dependencies in `pyproject.toml`. Otherwise `pip install .` to be safe?

Reply:
It requires a bit more finesse to do this in the pyproject.toml, because you'd need to avoid areas other than the test dependencies. I can double check that all pip installs are with `--build-isolation`.